README.rst

```diff
@@ -10,4 +10,174 @@ test base classes and documentation shared by many XBlocks.
+    ::
+        [
+            {"display_name": "Always", "value": "always"},
+            {"display_name": "Past Due", "value": "past_due"},
+        ]
+  - A range for floating point numbers with specific increments:
+    ``{"min": 0 , "max": 10, "step": .1}``
+  - A callable that returns one of the above. (Note: the callable does
+    *not* get passed the XBlock instance or runtime, so it cannot be a
+    normal member function)
+* ``values_provider`` can define a callable that accepts the XBlock
+  instance as an argument, and returns a list of possible values in one
+  of the formats listed above.
+* ``resettable_editor`` - defaults to ``True``. Set ``False`` to hide the
+  "Reset" button used to return a field to its default value by removing
+  the field's value from the XBlock instance.
+
+Basic screenshot: |Screenshot 1|
+
+StudioContainerXBlockMixin
+--------------------------
+
+.. code:: python
+
+    from xblockutils.studio_editable import StudioContainerXBlockMixin
+
+This mixin helps with creating XBlocks that want to allow content
+authors to add/remove/reorder child blocks. By removing any existing
+``author_view`` and adding this mixin, you'll get editable,
+re-orderable, deletable child support in Studio. To enable authors to
+add new children, simply override ``author_edit_view`` and set
+``can_add=True`` when calling ``render_children`` - see the source code.
+To enable authors to add only a limited subset of children requires
+custom HTML.
+
+An example is the mentoring XBlock: |Screenshot 2|
+
+SeleniumXBlockTest
+------------------
+
+.. code:: python
+
+    from xblockutils.base_test import SeleniumXBlockTest
+
+This is a base class that you can use for writing Selenium integration
+tests that are hosted in the XBlock SDK (Workbench).
+
+Here is an example:
+
+.. code:: python
+
+    class TestStudentView(SeleniumXBlockTest):
+        """
+        Test the Student View of MyCoolXBlock
+        """
+        def setUp(self):
+            super(TestStudentView, self).setUp()
+            self.set_scenario_xml('<mycoolblock display_name="Test Demo Block" field2="hello" />')
+            self.element = self.go_to_view("student_view")
+
+        def test_shows_field_2(self):
+            """
+            The xblock should display the text value of field2.
+            """
+            self.assertIn("hello", self.element.text)
+
+StudioEditableBaseTest
+----------------------
+
+.. code:: python
+
+    from xblockutils.studio_editable_test import StudioEditableBaseTest
+
+This is a subclass of ``SeleniumXBlockTest`` that adds a few helper
+methods useful for testing the ``studio_view`` of any XBlock using
+``StudioEditableXBlockMixin``.
+
+child\_isinstance
+-----------------
+
+.. code:: python
+
+    from xblockutils.helpers import child_isinstance
+
+If your XBlock needs to find children/descendants of a particular
+class/mixin, you should use
+``child_isinstance(self, child_usage_id, SomeXBlockClassOrMixin)``
```

Comment:
Please make this call and the next call separated out for ease of reading, eg

```
you should use

.. code:: python
    child_isinstance(self, child_usage_id, SomeXBlockClassOrMixin)

rather than calling

.. code:: python
    isinstance(self.runtime.get_block(child_usage_id), SomeXBlockClassOrMixin)
```

As-is, it's hard to read the rendered RST with all the code squashed together.